client/disputes/index.tsx

```diff
@@ -99,55 +102,65 @@ const headers = [
 	},
 ];
 
-export const DisputesList = () => {
+export const DisputesList = (): JSX.Element => {
 	const { disputes, isLoading } = useDisputes( getQuery() );
 
+	const {
+		disputesSummary,
+		isLoading: isSummaryLoading,
+	} = useDisputesSummary();
+
 	const rows = disputes.map( ( dispute ) => {
-		const order = dispute.order
-			? {
-					value: dispute.order.number,
-					display: <OrderLink order={ dispute.order } />,
-			  }
-			: null;
+		const {
+			amount = 0,
+			currency,
+			charge,
+			created,
+			id: disputeId,
+			evidence_details: evidenceDetails,
+			order: disputeOrder,
+			reason,
+			status,
+		} = dispute;
```

Comment:
I have multiple concerns here. 

The basis of them is of different structures that the client returns from the server. The cached disputes that are used for listing disputes differ from the proxied dispute that is used for details view. 

Firstly, the type hinting is incorrect. The cached dispute for example doesn't have `evidence_details`. This is going to be quite confusing so I'd suggest specifying in code that for listing disputes we use `CachedDispute` interface. 

Secondly, for the `evidence_details` example - this should not be present from the cached response. `id` also should be `dispute_id`.

Reply:
Ok, I found `format_dispute_for_ui` method that does the conversion in PHP side but there I still have concerns
1. The interface for Dispute is still not complete. To give example, details page takes the single dispute from list that was previously fetched and is missing required fields
2. Not sure if we want to drag the old structure forever into the codebase.